### ISSUE TITLE
Error on load image on React-native: Unexpected character

### ISSUE THREAD

**Issue by alexandref93** [facebook/react-native#6691]
react-native-cli: 0.1.10
react-native: 0.22.2
SO Mobile: Android
SO: Mac

I'm try display a image on a component on React-native, but I don't know why this error happens...

Example of code:

```
render () {
    let { convenience } = this.props
    return (
      <View style={{flexDirection: 'row', height: 50}}>
        <Text style={{marginRight: 30}}>{convenience.name}</Text>
        <Image source={require('./icons___favorito_ativo.png')} />
      </View>
    )
}
```

![](http://i.stack.imgur.com/FdZCI.png)

**Comment by seanfuture** [comment]
Get the same error but only when overwriting an image already loaded and cmd-r is pressed.  To resolve the issue temporarily it seems the image must be re-exported / overwritten and  then a 2nd cmd-r to reload results in clearing the error message.  Happens regularly at the moment for myself at least.  Haven't had time to research further.

**Comment by seanfuture** [comment]
Seems related to image map as noted here in this issue https://github.com/airbnb/enzyme/issues/253

**Comment by seanfuture** [comment]
Other potentials ( unlikely related to image map after all ):  
- https://github.com/RickWong/react-isomorphic-starterkit/issues/41
- https://github.com/gpbl/isomorphic500/issues/59

**Comment by sylvainbaronnet** [comment]
Same problem, fixed by re-exporting image as @seanfuture said

**Comment by sylvainbaronnet** [comment]
Restarting the packager fix it too

**Comment by dsibiski** [comment]
@facebook-github-bot answered

**Comment by seanfuture** [comment]
FYI - StackOverflow link:  http://stackoverflow.com/questions/36252381/error-on-load-image-on-react-native-unexpected-character/36289632#36289632 .. Don't believe this issue should have been closed as the above mentioned "fixes" aren't really fixes - Instead they're workarounds.  Error still comes up often and a full restart of the packager is needed.  Most definitely a prevalent and serious bug in the packager.  Taken a brief look, but packager code doesn't immediately reveal the cause although my suspicion is that Node.js is trying to process the file / file change as an asset.

**Comment by yashvekaria** [comment]
I too faced the same error. After a lot of trying, I restarted the packager, and the app picked up the image. So the solution is: Restart the packager.

Hope this helps.

**Comment by zipzit** [comment]
I know this is going sound pretty weird, but I'm going to add this comment in case anybody else gets here.  I created a index.ios.js file by copying a simple sample from an internet example (actually something at https://rnplay.org)  I kept getting "unexpected character" errors.  I'm using Atom.io as my script tool.  I was thinking perhaps I had an encoding issue with the wrong character set.  I've confirmed I'm using UTF-8

So I was using the (left/right) arrow keys on my keyboard, and I notice the cursor would stop moving for two keyboard arrow pushes, right at the location identified in my Emulator Red Screen of Disaster.  It was like there were two invisible characters in my code.  I played with this for a pretty long time to confirm.  I was able to highlight the "hidden" characters and delete them.  

After deletion, the new code works great.  

Bizarre.  Not sure what was there.  (Note: I copied the Slider Example Code from https://rnplay.org/apps/5FsfPA and I used a "Select All" and Command-C to do the copying and Command-V to paste... if anybody wants to repeat the experiment)   

And yes, I know how silly this sounds.  Perhaps others have hit the same roadbump?

**Comment by duzliang** [comment]
stop the server and execute `react-native start` again

**Comment by jaredly** [comment]
the packager should not require this though -- this *is a bug*, and even though there's a workaround, it should be fixed. this isn't a SO type question, it's a "please fix the packager"

**Comment by tomchambers2** [comment]
Still having this issue. Happened in various occasions, most reliably when I open an image being used in the react native app with preview (hit space bar in finder) on OSX.

**Comment by bitraven** [comment]
I'm having the same issue, reproduced with a stock install using only the first block of sample code in the React-Native docs for the Image component here:  https://facebook.github.io/react-native/docs/image.html, reproduced here:

`import React, { Component } from 'react';
import { AppRegistry, View, Image } from 'react-native';

class DisplayAnImage extends Component {
  render() {
    return (
      <View>
        <Image
          source={require('./img/favicon.png')}
        />
        <Image
          style={{width: 50, height: 50}}
          source={{uri: 'https://facebook.github.io/react/img/logo_og.png'}}
        />
      </View>
    );
  }
}`

restarting the packager does not resolve the issue.  (I wasn't sure if I should make a new question, referencing this one, or just add on to this one ... I went with adding on, since there is a lot of context here)

if I comment out the first Image block (the local file require), then the second one (remote file) works fine.

I'm attempting to learn/eval react-native, so I don't yet have a deep understanding of the system or what is relevant info to provide here that would help debug the issue.  I can say it's a stock, clean installation following the Getting Started document (https://facebook.github.io/react-native/docs/getting-started.html) and using only the previously mentioned example, also from the docs.

In order to get even that far, I did have to comment out a similar require in react-native/Libraries/CustomComponents/NavigationExperimental/NavigationHeaderBackButton.js, in the NavigationHeaderBackButton, where it tried to require a local asset 'back-icon.png', which was causing the same problem the sample is now exhibiting.

I'm willing to follow up and provide any information which might be useful, if anyone has some insight!

**Comment by felipecocco** [comment]
+1

**Comment by F1ks3r** [comment]
Re exporting image, re encoding it, different encodings, sizes, problem remains...

**Comment by F1ks3r** [comment]
Switching react-native from 40 to 38 helped, i don't think this should have happen

**Comment by F1ks3r** [comment]
Another suggestion is to `watchman watch-del-all` and running the RN packager server with the --resetCache option.

**Comment by radelcom** [comment]
started seeing this today after updating to RN v0.40.0. Tried all the workarounds mentioned in this thread but no luck 😞

**Comment by fdb** [comment]
I have this issue as well. Turns out it is because I have **spaces in the project path**:

- This works: `/Users/fdb/Desktop/myproject/icons/settings.png`
- This doesn't: `/Users/fdb/Dropbox (Personal)/myproject/icons/settings.png`

Running `watchman watch-del-all` has no effect — only moving to a path without spaces does.

**Comment by mrlords1** [comment]
I had this same error, 
When i try stop n start server see this log:

myfile.png: Unexpected character '�' (1:0)
> 1 | �PNG
    | ^
  2 | 
  3 | 
  4 | IHDR��X��d�IT   pHYs.#.#x�?vtEXtSoftwareAdobe Fireworks CS6輲�tEXtCreation Time11/14/166z�� IDATx��}w�dU��{��\��'�0PMRXP�"
                                                                                                                                     +�JPW@u
soo I try user randon .PNG google image, and it work. 

After I need re export my file.

**Comment by joeferraro** [comment]
@fdb I **do not** have spaces in my project path and still experience this issue.

**Comment by radelcom** [comment]
@fdb that seems to be the case for me as well.

**Comment by holgersindbaek** [comment]
The space in the path was the problem for me as well. I can't change that space, since it's controlled by Dropbox. Any other solution to this issue?

**Comment by cyclops24** [comment]
Hi guys,
I also want to create simple hello word based on this: https://facebook.github.io/react-native/docs/getting-started.html and I faced on this red error:
```
"node_modules/react-native/Libraries/CustomComponents/NavigationExperimental/assets/back-icon.png doesn't exist"
```

Can you help me to fix this?
I worked with React-Native later without any issue but after install new version this error appear on any app (also a simple hello world)

**Comment by cyclops24** [comment]
This is my complete error:
![photo_2017-03-04_20-51-18](https://cloud.githubusercontent.com/assets/8266479/23580755/d036e788-011c-11e7-9a2a-b2801d31a2b1.jpg)

**Comment by joshfriend** [comment]
@cyclops24 restart the packager. It doesn't seem to pick up new images automatically.

**Comment by cyclops24** [comment]
@joshfriend how restart packager. It's a simple hello world without any image:
I just run this command:
```
react-native init AwesomeProject cd AwesomeProject react-native run-android
```
Exactly based on https://facebook.github.io/react-native/docs/getting-started.html

**Comment by LoekL** [comment]
Here's how you restart packager: https://github.com/drapanjanas/re-natal/issues/56

I still don't get it to work though, re-starting packager nor path name without spaces fix it.

**Comment by a1exlism** [comment]
I found a phenomenon that Every file renaming, adding, editing or deleting , I always should `npm start` and `cmd r`to receive file again.That's the way what I resolved this prob.

**Comment by brentvatne** [comment]
I've confirmed that this occurs when a path has parens, which is why people using Dropbox in this thread have this issue -- `Dropbox (Personal)` is part of the path. Spaces are working fine. cc @mkonicek

**Comment by jan-happy** [comment]
The original topic of this issue is the "Unexpected character" bug.
It always occurs when an image file is added or changed.

**Comment by woshi82** [comment]
when i  integration-with-existing-apps the image cannot show . the problem is same with https://github.com/facebook/react-native/issues/12062. and restart package not work.  `watchman watch-del-all` not work

**Comment by cyclops24** [comment]
@woshi82 I also have `+` in project parent name and after remove it everything goes OK.
Is it a bug or `+` is not acceptable for project folder name (maybe an undocumented limitation). I didn't find info or limitation about React-Native project name or path in docs.

**Comment by mitchjam** [comment]
I had a ```*``` in my folder path and after I removed it my error was gone.

**Comment by agersoncgps** [comment]
I have no special chaecters in my path and it happens to me any time I make a change to the image and re-save it.

**Comment by silkyland** [comment]
This work for me 
just type "react-native link" in terminal 
The reason is because you create new assets folder in your project and react-native may not know it

sorry for my bad English

**Comment by idrisdopico** [comment]
Not using dropbox, don't have a `+` in the code. When restarting the terminal and simulator it works fine, but like others have said, this is a workaround. Not a fix.

**Comment by dcu-sharepoint** [comment]
as suggested by many here; restarted the entire process environment and now everything works. However, this is a BUG that NEEDS FIXING without any workarounds if this is going to make to the real world. Going to SO with a question like brings out the arrogant admins where they take GREAT pleasure telling how to ask intelligent questions and what is an intelligent question. To them, this IS NOT and intelligent question so no answer is given. I speak from frustrating experiences with SO, so I've learned to read, not speak.

**Comment by spenghang** [comment]
still having this issue, and  restart doesn't work for me. RN 0.44.

**Comment by jhalborg** [comment]
Restart of packager works, but is a bit annoying :-)

**Comment by deathemperor** [comment]
Just got this on 0.44.2. my path has '-' not sure if it's the reason. Restarting the packager fixed.

**Comment by muhammadridho** [comment]
I have a same problem, because my image's name has '-','+','_'. the solution is try react-native link or restart packager.

**Comment by zakster12** [comment]
I have very easy fix for the problem that works perfectly for me. I remove the problematic image from the assets folder and add it again. And the red screen is gone. Just drag & drop.

**Comment by SudoPlz** [comment]
Removing spaces and parenthesis from my path fixed that issue for me too.

**Comment by mr-bat** [comment]
I solved(!?) this problem by creating a copy of the image and replacing the original one, then renamed the image. This indeed is a weird bug, your photo at once becomes useless.

**Comment by ShubhamOjha** [comment]
Closing the server and restarting it works just fine.

**Comment by Rebaiahmed** [comment]
@sylvainbaronnet thanks it's working for me :+1:

**Comment by jan-happy** [comment]
@ShubhamOjha: I don't think this is really a solution. One nice thing about react native is the hot reloading feature. Which is made completely useless if you use images in your project that change during development.

**Comment by milon87** [comment]
faced the same problem and solved by restarting the project

**Comment by paranoidjk** [comment]
same problem here, any process?

**Comment by gblaender** [comment]
Same problem, but nothing worked. Killed and restarted the packager, replaced the image... Same unexpected character red screen. Really frustrating.. Newbie dev here :)

**Comment by cpojer** [comment]
This is fixed for the next release of react-native, where we have many HMR improvements. cc @grabbou can tell you when that goes out but in your current version of react-native it isn't magically going to be fixed from one minute to the next.

**Comment by danilags** [comment]
Try terminate server your terminal and react-native run-android / ios  🗡

**Comment by songming43** [comment]
As @danilags says  

- Try terminate your packager server
- `rm -fr $TMPDIR/react-*` clear the react-native cache
- rerun your xcode  
hope this helps you

**Comment by jgreen01** [comment]
tl;dr Always use `.png` not `.PNG`

I had this same issue. Using an uppercase `PNG` file extension will also case this error.

**Comment by norby3** [comment]
foobar.JPG  causes this really annoying error
foobar.jpg  works 

.. lotta people wasted a lot of time .. not a confidence booster
